common-custom-user-data-maven-extension/README.md

```diff
@@ -30,7 +44,26 @@ This approach has a number of benefits:
 - Tailor the build scan enhancements to exactly the set of tags, links and custom values you require.
 - Standardize the configuration for connecting to Gradle Enterprise and Gradle Build Cache in your organization, removing the need for each project to specify this configuration.
 
-If your customized extension provides all required Gradle Enterprise configuration, then a consumer project will get all the benefits of Gradle Enterprise simply by applying the extension. The plugin sources provide a placeholder and example code to get you started.
+The project sources provide a good template to get started with your own Maven extension. 
+Refer to the [Javadoc](https://docs.gradle.com/enterprise/maven-extension/api/) for more details on the key types available for use.
+
+#### Providing Gradle Enterprise configuration in your custom Maven extension
+
+Using a custom extension, it is possible to completely remove the need for each project to declare a `gradle-enterprise.xml` file.
+Much of the configuration can be supplied programmatically. See `CustomBuildScanConfig` and `CustomBuildCacheConfig` for examples.
+
+At this time, the Gradle Enterprise server URL and credentials cannot be provided programmatically, and must be specified in a `gradle-enterprise.xml` file.
+To avoid adding this file to every project, this configuration file can be added to the `src/main/resources` directory so that it will be discovered by Gradle Enterprise when loading the extension.
```

Comment:
I would make it a bit more explicit that putting the file into src/main/resources means it will be packaged in the custom extension and discovered when loading the Gradle Enterprise extension.